packages/@react-aria/selection/src/useSelectableCollection.ts

```diff
@@ -487,14 +488,19 @@ export function useSelectableCollection(options: AriaSelectableCollectionOptions
       if (focusedKey == null && !shouldUseVirtualFocus && ref.current) {
         focusSafely(ref.current);
       }
+
+      // Wait until the collection has items to autofocus.
```

Comment:
Previously we delayed rendering the menu until the collection had items, but we can't do this anymore now that we have empty state support. Instead, delay auto focusing until the collection is filled.